src/components/VHeader/VHeaderInternal.vue

```diff
@@ -0,0 +1,137 @@
+<template>
+  <header
+    class="main-header z-30 flex h-20 w-full items-stretch justify-between gap-x-2 bg-white py-4 pe-3 ps-6 md:py-4 md:px-7"
+  >
+    <VBrand :is-fetching="false" />
+    <nav class="justify-stretch md:justify-stretch hidden ms-auto md:flex">
+      <ul class="flex flex-row items-center gap-8 text-sm">
+        <li v-for="page in allPages" :key="page.id">
+          <VLink
+            class="text-dark-charcoal"
+            :class="currentPage === page.id ? 'font-bold' : ''"
+            :href="page.link"
+            show-external-icon
+            >{{ $t(`navigation.${page.id}`) }}</VLink
+          >
+        </li>
+      </ul>
+    </nav>
+    <nav class="flex md:hidden">
+      <VModal
+        :label="$t('header.aria.menu').toString()"
+        variant="full"
+        mode="dark"
+        :visible="isModalVisible"
+        @open="setModalVisibility(true)"
+      >
+        <template #trigger="{ a11yProps }">
+          <VIconButton
+            ref="menuButtonRef"
+            :icon-props="{ iconPath: menuIcon }"
+            :button-props="{ variant: 'plain' }"
+            v-bind="a11yProps"
+            class="border-tx"
+            size="search-medium"
+          />
+        </template>
+
+        <template #top-bar>
+          <div
+            class="flex h-20 w-full justify-between bg-dark-charcoal py-4 text-white pe-3 ps-6"
+          >
+            <VBrand :is-fetching="false" />
+            <VIconButton
+              ref="closeButton"
+              variant="plain"
```

Comment:
This does not appear to be an actual prop on `VIconButton`. Is this serving a different function here?

```suggestion
```